src/Faithlife.Testing/AssertEx.cs

```diff
@@ -842,6 +836,69 @@ private static string ToString(object obj)
 			return toString;
 		}
 
+		private static string ToPrettyJson(object obj, int tabLevel)
+		{
+			var rawJson = JsonUtility.ToJson(obj, s_jsonWithTabs).Replace("\r\n", "\n");
+			var lines = rawJson.Split('\n').Select(l => l.Trim()).ToList();
+
+			// if json can fit on one line, return inlined
+			var shortJson = lines.Select(l => l + (l.EndsWithOrdinal(",") ? " " : "")).Join("");
+			if (shortJson.Length < c_maxJsonLength)
+				return shortJson;
+
+			// else construct pretty json output
+			var longJson = "";
```

Comment:
Nit: Whenever you're doing a whole bunch of arbitrarily-large string operations, it's nice to use `StringBuilder`.

We're already using expression trees in this code, so you could reasonably argue that's a micro-optimization.

But I find that `StringBuidler` makes the code read more clearly, too, by emphasizing the your intent better,

Reply:
Ooh, I'm not super familiar with StringBuilder. I'll look into it.

Reply:
After implementing Bradley's suggestion, I'm not sure if this advice is still relevant